fontra2fontir/src/toir.rs

```diff
@@ -0,0 +1,298 @@
+                .map_err(WorkError::PathConversionError)?,
+            PointType::OffCurveQuad | PointType::OffCurveCubic => path_builder
+                .offcurve((point.x, point.y))
+                .map_err(WorkError::PathConversionError)?,
+        }
+    }
+    Ok(())
+}
+
+fn to_ir_path(glyph_name: GlyphName, contour: &FontraContour) -> Result<BezPath, WorkError> {
+    // Based on glyphs2fontir/src/toir.rs to_ir_path
+    // TODO: so similar a trait to to let things be added to GlyphPathBuilder would be nice
+    if contour.points.is_empty() {
+        return Ok(BezPath::new());
+    }
+
+    let mut path_builder = GlyphPathBuilder::new(glyph_name.clone(), contour.points.len());
+
+    if !contour.is_closed {
+        let first = contour.points.first().unwrap();
+        let first_type = first
+            .point_type()
+            .map_err(|e| WorkError::InvalidSourceGlyph {
+                glyph_name: glyph_name.clone(),
+                message: format!("No point type for {first:?}: {e}"),
+            })?;
+        if first_type.is_off_curve() {
+            return Err(WorkError::InvalidSourceGlyph {
+                glyph_name: glyph_name.clone(),
+                message: String::from("Open path starts with off-curve points"),
+            });
+        }
+        path_builder.move_to((first.x, first.y))?;
+        add_to_path(
+            glyph_name.clone(),
+            &mut path_builder,
+            contour.points[1..].iter(),
+        )?;
+    } else {
+        add_to_path(glyph_name.clone(), &mut path_builder, contour.points.iter())?;
+    }
+
+    let path = path_builder.build()?;
+    trace!(
+        "Built a {} entry path for {}",
+        path.elements().len(),
+        glyph_name
+    );
+    Ok(path)
+}
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+
+    use fontdrasil::{coords::NormalizedCoord, types::Axis};
+    use fontir::ir::Glyph;
+    use kurbo::{BezPath, PathEl};
+    use write_fonts::types::Tag;
+
+    use crate::{
+        fontra::{FontraFontData, FontraGlyph},
+        test::testdata_dir,
+        toir::to_ir_static_metadata,
+    };
+
+    use super::to_ir_glyph;
+
+    fn axis_tuples(axes: &[Axis]) -> Vec<(&str, Tag, f64, f64, f64)> {
+        axes.iter()
+            .map(|a| {
+                (
+                    a.name.as_str(),
+                    a.tag,
+                    a.min.to_f32() as f64,
+                    a.default.to_f32() as f64,
+                    a.max.to_f32() as f64,
+                )
+            })
+            .collect::<Vec<_>>()
+    }
+
+    fn commands(b: &BezPath) -> String {
+        b.elements()
+            .iter()
+            .map(|e| match e {
+                PathEl::MoveTo(..) => "M",
+                PathEl::LineTo(..) => "L",
+                PathEl::QuadTo(..) => "Q",
+                PathEl::CurveTo(..) => "C",
+                PathEl::ClosePath => "Z",
+            })
+            .collect::<Vec<_>>()
+            .join("")
```

Comment:
```suggestion
            .map(|e| match e {
                PathEl::MoveTo(..) => 'M',
                PathEl::LineTo(..) => 'L',
                PathEl::QuadTo(..) => 'Q',
                PathEl::CurveTo(..) => 'C',
                PathEl::ClosePath => 'Z',
            })
            .collect()
```
Tiny little trick: you can collect an iterator of `char` directly into a `String`.